Core_Framework/DriverCore/WebDriverBase.cs

```diff
@@ -0,0 +1,569 @@
+    public bool IsElementDisplay(string locator)
+    {
+        IWebElement e = Driver.FindElement(GetXpath(locator));
+
+        if (e == null)
+        {
+            HtmlReport.Fail("Element is not displayed");
+            return false;
+        }
+        else
+        {
+            HighlightElement(e);
+            HtmlReport.Pass("Element [" + e.ToString() + "] is displayed", TakeScreenShot());
+            return true;
+        }
+    }
+
+    public bool IsErrorMessageDisplay(string locator)
+    {
+        IWebElement errorMessage = Driver.FindElement(GetXpath(locator));
+
+        if (errorMessage == null)
+        {
+            HtmlReport.Fail("Errormessage is not displayed");
+            return false;
+        }
+        else
+        {
+            HighlightElement(errorMessage);
+            Console.Write(errorMessage.Text);
+            Assert.IsTrue(errorMessage.Text.Contains("Username or password is incorrect!"));
+            HtmlReport.Pass("Errormessage [" + errorMessage.Text + "] is displayed", TakeScreenShot());
+            return true;
+        }
+    }
+
+    public bool IsPopUpDisplay(string locator)
+    {
+        IWebElement e = Driver.FindElement(GetXpath(locator));
+
+        if (e == null)
+        {
+            HtmlReport.Fail("Popup does not display");
+            return false;
+        }
+        else
+        {
+            HighlightElement(e);
+            HtmlReport.Pass("Popup displays", TakeScreenShot());
+            return true;
+        }
+    }
+
+    public void CompareTitles(string expectedTitle)
+    {
+        try
+        {
+            string actualTitle = GetTitle();
+            Assert.That(actualTitle, Is.EqualTo(expectedTitle));
+            HtmlReport.Pass("Actual title [" + actualTitle + "] matches [" + expectedTitle + "]", TakeScreenShot());
+        }
+        catch (Exception)
+        {
+            HtmlReport.Fail("Titles do not match", TakeScreenShot());
+            throw;
+        }
+    }
+
+    public void CompareUrls(string expectedUrl)
+    {
+        try
+        {
+            string actualUrl = GetUrl();
+            Assert.That(actualUrl, Is.EqualTo(expectedUrl));
+            HtmlReport.Pass("Actual Url [" + actualUrl + "] matches [" + expectedUrl + "]", TakeScreenShot());
+        }
+        catch (Exception)
+        {
+            HtmlReport.Fail("Urls do not match", TakeScreenShot());
+            throw;
+        }
+    }
+
+    public static void AssertEquals(object actual, object expected)
+    {
+        try
+        {
+            actual.Should().BeEquivalentTo(expected);
+            HtmlReport.Pass("Actual data [" + actual.ToString() + "] matches " +
+                "with expected data [" + expected.ToString() + "]");
+        }
+        catch (Exception excep)
+        {
+            HtmlReport.Fail("Actual data [" + actual + "] does not match " +
+                "with expected data [" + expected + "]");
+            throw excep;
+        }
+    }
```

Comment:
This method does not belong to WebDriverbase. Please move it to another object like CustomAssertion or smt like that